ios/RNSnowplowTracker.m

```diff
@@ -433,6 +435,148 @@ @implementation RNSnowplowTracker
+
+        NSString *title = [argmap sp_stringForKey:@"title" defaultValue:nil];
+        NSString *body = [argmap sp_stringForKey:@"body" defaultValue:nil];
+        NSString *triggerStr = [argmap sp_stringForKey:@"trigger" defaultValue:nil];
+        SPMessageNotificationTrigger trigger;
+        if ([triggerStr isEqualToString:@"push"]) {
+            trigger = SPMessageNotificationTriggerPush;
+        } else if ([triggerStr isEqualToString:@"location"]) {
+            trigger = SPMessageNotificationTriggerLocation;
+        } else if ([triggerStr isEqualToString:@"calendar"]) {
+            trigger = SPMessageNotificationTriggerCalendar;
+        } else if ([triggerStr isEqualToString:@"timeInterval"]) {
+            trigger = SPMessageNotificationTriggerTimeInterval;
+        } else {
+            trigger = SPMessageNotificationTriggerOther;
+        }
+        SPMessageNotification *event = [[SPMessageNotification alloc] initWithTitle: title
+                                                                               body: body
+                                                                            trigger: trigger];
+
+        NSString *action = [argmap sp_stringForKey:@"action" defaultValue:nil];
+        if (action) {
+            event.action = action;
+        }
+        NSArray *attachmentsMap = [argmap objectForKey:@"attachments"];
+        if (attachmentsMap) {
+            NSMutableArray *attachments = [NSMutableArray new];
+            for (NSDictionary* attachmentMap in attachmentsMap) {
+                NSString *identifier = [attachmentMap sp_stringForKey:@"identifier" defaultValue:nil];
+                NSString *type = [attachmentMap sp_stringForKey:@"type" defaultValue:nil];
+                NSString *url = [attachmentMap sp_stringForKey:@"url" defaultValue:nil];
+                SPMessageNotificationAttachment *attachment = [[SPMessageNotificationAttachment alloc] initWithIdentifier:identifier
+                                                                                                                     type:type
+                                                                                                                      url:url];
+                [attachments addObject:attachment];
+            }
+            event.attachments = attachments;
+        }
+        NSArray<NSString *> *bodyLocArgs = [argmap objectForKey:@"bodyLocArgs"];
+        if (bodyLocArgs) {
+            event.bodyLocArgs = bodyLocArgs;
+        }
+        NSString *bodyLocKey = [argmap sp_stringForKey:@"bodyLocKey" defaultValue:nil];
+        if (bodyLocKey) {
+            event.bodyLocKey = bodyLocKey;
+        }
+        NSString *category = [argmap sp_stringForKey:@"category" defaultValue:nil];
+        if (category) {
+            event.category = category;
+        }
+        NSNumber *contentAvailable = [argmap sp_numberForKey:@"contentAvailable" defaultValue:nil];
```

Comment:
not sure how this plays out with booleans, how about `sp_boolForKey`?

Reply:
NSNumber also supports booleans. We could also use `sp_boolForKey` but the `event.contentAvailable` property takes NSNumber so we would have to transform to NSNumber anyway.